### PR TITLE
[New Rule] Creation of a Hidden Launch Agent or Daemon

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/796

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/macos/persistence_evasion_hidden_launch_agent_deamon_creation.toml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+type = "eql"
+
+query = '''
+file where event.type != "deletion" and
+  wildcard(file.path,
+            "/System/Library/LaunchAgents/.*.plist",
+            "/Library/LaunchAgents/.*.plist",
+            "/Users/*/Library/LaunchAgents/.*.plist",
+            "/System/Library/LaunchDaemons/.*.plist",
+            "/Library/LaunchDaemons/.*.plist")
```
Comment (bm11100) [inline rules/macos/persistence_evasion_hidden_launch_agent_deamon_creation.toml:33]:
```suggestion
file where event.type != "deletion" and
  file.path:
  (
    "/System/Library/LaunchAgents/.*.plist",
    "/Library/LaunchAgents/.*.plist",
    "/Users/*/Library/LaunchAgents/.*.plist",
    "/System/Library/LaunchDaemons/.*.plist",
    "/Library/LaunchDaemons/.*.plist"
  )
```

Comment (bm11100) [reply rules/macos/persistence_evasion_hidden_launch_agent_deamon_creation.toml:33]: spacing and wildcard function removal - 
```
file where event.type != "deletion" and
  file.path:
  (
    "/System/Library/LaunchAgents/.*.plist",
    "/Library/LaunchAgents/.*.plist",
    "/Users/*/Library/LaunchAgents/.*.plist",
    "/System/Library/LaunchDaemons/.*.plist",
    "/Library/LaunchDaemons/.*.plist"
  )
```

File: rules/macos/persistence_evasion_hidden_launch_agent_deamon_creation.toml [hunk 2]
```diff
@@ -0,0 +1,63 @@
+[metadata]
+creation_date = "2020/01/05"
+maturity = "production"
+updated_date = "2020/01/05"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the creation of hidden launch agent or daemon. An adversary can establish persistence by installing a new
+launch agent or daemon that executes at login.
```
Comment (bm11100) [inline rules/macos/persistence_evasion_hidden_launch_agent_deamon_creation.toml:10]:
```suggestion
Identifies the creation of a hidden launch agent or daemon. An adversary may establish persistence by installing a new
launch agent or daemon which executes at login.
```

File: rules/macos/persistence_evasion_hidden_launch_agent_deamon_creation.toml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "eql"
+license = "Elastic License"
+name = "Creation of a Hidden Launch Agent or Daemon"
```
Comment (bm11100) [inline rules/macos/persistence_evasion_hidden_launch_agent_deamon_creation.toml:16]:
```suggestion
name = "Creation of Hidden Launch Agent or Daemon"
```

File: rules/macos/persistence_evasion_hidden_launch_agent_deamon_creation.toml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+]
+risk_score = 47
+rule_id = "092b068f-84ac-485d-8a55-7dd9e006715f"
+severity = "medium"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Persistence", "Defense Evasion"]
+type = "eql"
+
+query = '''
+file where event.type != "deletion" and
+  file.path:
```
Comment (bm11100) [inline rules/macos/persistence_evasion_hidden_launch_agent_deamon_creation.toml:28]:
```suggestion
  file.path : 
```

**Review by bm11100 — APPROVED** [review]

**Review by threat-punter — APPROVED** [review]
😎

**Pull request merged**